api/v1/networkplugins_types.go

```diff
@@ -28,10 +28,37 @@ type NetworkPluginsSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of NetworkPlugins. Edit NetworkPlugins_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	CniPlugins *CniPlugins `json:"cniPlugins,omitempty"`
+
+	//HostNetworking *HostNetworking `json:hostNetworking,omitempty"`
+}
+
+type CniPlugins struct {
+	Multus      *Multus      `json:"multus,omitempty"`
+	Whereabouts *Whereabouts `json:"whereabouts,omitempty"`
+	Sriov       *Sriov       `json:"sriov,omitempty"`
+	NMState     *NMState     `json:"nmstate,omitempty"`
+}
+
+type Whereabouts struct {
```

Comment:
Can you add a description to these structs even if it is one line or just a link, to understand what it is modeling.

Reply:
(for each of the structs)